src/engine_redraw.c

```diff
@@ -518,6 +518,23 @@ void redraw_creature_view(void)
     message_draw();
     gui_draw_all_boxes();
     draw_tooltip();
+    long x = 148 * units_per_pixel / 16;
+    static const SpellKind Spells[] = {SplK_Freeze, SplK_Armour, SplK_Rebound, SplK_Invisibility, SplK_Speed, SplK_Slow, SplK_Light, SplK_Fly, SplK_Sight, SplK_Disease, SplK_Chicken};
```

Comment:
Is this list here needed? Does creature_affected_by_spell not handle by which spells you are affected?

Reply:
We don't need to check every spell, surely?

Reply:
Need is a big word. Having hardcoded lists like this is asking for trouble in the future when seemingly unrelated stuff changes. And it's not a heavy check at all, it is only for the one possessed unit, and the function that checks it returns false on the irrelevant spells right away.